### DISCUSSION TITLE
Firmware problème

### DISCUSSION THREAD

**Discussion by GAKFANDANG** [micropython/micropython#14414] [ESP32]
It's already been a week since I downloaded the micropythons for my ESP32-CAM board but I never come across the correct ones, please can someone help me with the correct one?

**Comment by andrewleech** [comment]
Which firmware did you download, I would suggest the generic esp32 one in spiram variant should be correct?

https://micropython.org/download/ESP32_GENERIC/

Scroll down into the section "Firmware (Support for SPIRAM / WROVER)"

**Reply by davefes** [reply]
I think for camera operation you need something like:
https://github.com/shariltumin/esp32-cam-micropython-2022/tree/main/X23
and then pick the board you use.

I use:
https://github.com/shariltumin/esp32-cam-micropython-2022/tree/main/X23/esp32-aiThinker/firmwares/wifi%2Bssl
